src/nats.h

```diff
@@ -1281,26 +1281,39 @@ typedef struct jsOptions
 
         struct jsOptionsSubscribePullAsync
         {
-                // Options to control automatic Fetch flow control.
-                //
-                // The number of messages to ask for in a single request, and if
-                // we should try to fetch ahead, KeepAhead more than we need to
-                // finish the current request. Fetch this many messages ahead of
-                // time.
-                int FetchSize;
-                int KeepAhead;
+                // Lifetime of the subscription (completes when any one of the
+                // targets is reached).
+                int64_t                 TimeoutMillis;
+                int                     MaxMessages;
+                int64_t                 MaxBytes;
 
-                // Manual fetch flow control. If provided gets called before
-                // each message is deliverered to msgCB, and overrides the
-                // default algorithm for sending Next requests.
-                natsFetchNextHandler NextHandler;
-                void *NextHandlerClosure;
+                // If NoWait is set, the subscription will receive the messages
+                // already stored on the server subject to the limits, but will
+                // not wait for more messages.
+                bool                    NoWait;
 
                 // Fetch complete handler that receives the exit status code,
                 // the subscription's Complete handler is also invoked, but does
                 // not have the status code.
                 natsFetchCompleteHandler CompleteHandler;
-                void *CompleteHandlerClosure;
+                void                    *CompleteHandlerClosure;
+
+                // Have server sends heartbeats to help detect communication failures.
+                int64_t                 HeartbeatMillis;
```

Comment:
nit: Same here. For `Heartbeat`, if you feel that it is not telling enough, it could have been `HeartbeatInterval`.